mono/metadata/object-internals.h

```diff
@@ -25,7 +25,7 @@
 #define MONO_CHECK_ARG(arg, expr, retval) do {				\
 	if (G_UNLIKELY (!(expr)))					\
 	{								\
-		if (arg) {} /* check if the name exists */		\
+		(void)(arg); /* check if the name exists */		\
```

Comment:
(bikeshedding) why not `if (0) { (void)(arg); }` ?

Reply:
Seems like extra unnecessary evaluation.

Reply:
1. Is it good or bad that the old form is invalid for type safe handles?
I forsee it being valid in C++ though, to replace MONO_HANDLE_IS_NULL.

2. I misread your comment. I thought you were asking vs. what was there.
Your ```if (0) { (void)(arg); }`` is nice.